modules/_labs/side-panel/react/lib/SidePanel.tsx

```diff
@@ -0,0 +1,256 @@
+      width: ${normalized.from};
+      min-width: ${normalized.from};
+      max-width: ${normalized.from};
+    } to {
+      width: ${normalized.to};
+      min-width: ${normalized.to};
+      max-width: ${normalized.to};
+    }
+  `;
+};
+
+const containerVariantStyle: {[K in SidePanelVariant]: CSSObject} = {
+  alternate: {
+    backgroundColor: colors.frenchVanilla100,
+    ...depth[3],
+  },
+  standard: {
+    backgroundColor: colors.soap100,
+  },
+};
+
+const Panel = styled('section')<Pick<SidePanelProps, 'as'>>({
+  overflow: 'hidden',
+  position: 'relative',
+  boxSizing: 'border-box',
+});
+
+export const SidePanelContext = React.createContext({
+  state: 'expanded',
+  origin: 'left',
+});
+
+const SidePanel = ({
+  as = 'section',
+  children,
+  collapsedWidth = 64,
+  expanded = true,
+  expandedWidth = 320,
+  height = '100%',
+  onAnimationEnd,
+  onAnimationStart,
+  onExpandedChange,
+  onStateTransition,
+  origin = 'left',
+  variant = 'standard',
+  ...elemProps
+}: SidePanelProps) => {
+  const mounted = React.useRef(false);
+  const [state, setState] = React.useState<SidePanelTransitionStates>(
+    expanded ? 'expanded' : 'collapsed'
+  );
+
+  // This is meant to prevent animations when the component renders for the first time.
+  // mounted.current will only be false on the first pass
+  React.useEffect(() => {
+    mounted.current = true;
+  }, []);
+
+  React.useEffect(() => {
+    if (typeof onExpandedChange !== 'undefined') {
+      onExpandedChange(expanded);
+    }
+  }, [expanded, onExpandedChange]);
+
+  React.useEffect(() => {
+    if (typeof onStateTransition !== 'undefined') {
+      onStateTransition(state);
+    }
+  }, [state, onStateTransition]);
+
+  const motion = {
+    collapse: createKeyframes(expandedWidth, collapsedWidth),
+    expand: createKeyframes(collapsedWidth, expandedWidth),
+  };
+
+  const handleAnimationEnd = (event: React.AnimationEvent<HTMLDivElement>) => {
+    if (event.currentTarget === event.target) {
+      if (
+        event.animationName === motion.collapse.name ||
+        event.animationName === motion.expand.name
+      ) {
+        setState(expanded ? 'expanded' : 'collapsed');
+      }
+    }
+
+    if (typeof onAnimationEnd !== 'undefined') {
+      onAnimationEnd(event);
+    }
+  };
+
+  const handleAnimationStart = (event: React.AnimationEvent<HTMLDivElement>) => {
+    if (event.currentTarget === event.target) {
+      if (
+        event.animationName === motion.collapse.name ||
+        event.animationName === motion.expand.name
+      ) {
+        setState(expanded ? 'expanding' : 'collapsing');
+      }
+    }
+
+    if (typeof onAnimationStart !== 'undefined') {
+      onAnimationStart(event);
+    }
+  };
+
+  return (
+    <Panel
+      as={as}
+      css={[
+        {
+          width: expanded ? expandedWidth : collapsedWidth,
+          maxWidth: expanded ? expandedWidth : collapsedWidth,
+          minWidth: expanded ? expandedWidth : collapsedWidth,
```

Comment:
It was probably due to me screwing around with height and widths that are not numerical px values. However, in the spirit of not boiling the ocean...

`collapsedWidth` and `expandedWidth` now map to `minWidth` and `width` only.

As discussed behind-the-scenes, `height` will now just be `height: 100%` without `min` or `max` specified, the user can dictate whether or not they want it to overflow or grow the container that it is in (stories have been updated to reflect this). We won't force a preference here. Also, I removed the `height` prop, it can just be set via `style` or emotion's `css` prop.